### PR TITLE
fix: properly clear value in Form.clear_value/2

### PR DESCRIPTION
**Pull request by minibikini**
### Contributor checklist

- [x] Bug fixes include regression tests

### CODE REVIEW

File: test/form_test.exs [hunk 1]
```diff
@@ -99,9 +99,18 @@ defmodule AshPhoenix.FormTest do
         |> Form.validate(%{"text" => "text"})
 
       assert Form.value(form, :text) == "text"
+      assert form.source.arguments == %{}
```
[the remark this answers is not in the document]
Comment (zachdaniel) [reply test/form_test.exs:102]: You could add a text argument to the relevant action that is being run: https://github.com/ash-project/ash_phoenix/blob/main/test/support/resources/post.ex#L18 and then try doing `clear_value` on it.

[the remark this answers is not in the document]
Comment (minibikini) [reply test/form_test.exs:102]: Thanks, will try now.

[the remark this answers is not in the document]
Comment (minibikini) [reply test/form_test.exs]: When I added `text` argument, it disappeared from the `attributes` map, so I've added a new optional `excerpt` argument and a new almost identical test for it. Hope that's okay. Please, review.

**Review by zachdaniel — APPROVED** [review]

**Comment by zachdaniel** [comment]
🚀 Thank you for your contribution! 🚀

**Pull request merged**